rootfs/api/models/certificate.py

```diff
@@ -202,13 +202,16 @@ def detach(self, *args, **kwargs):
 
         name = '%s-cert' % self.name
         app = domain.app
-        # only delete if it exists
-        try:
-            # We raise an exception when a secret doesn't exist
-            self._scheduler._get_secret(app, name)
-            self._scheduler._delete_secret(app, name)
-        except KubeHTTPException as e:
-            logger.critical(e)
+
+        # only delete if it exists and if no other domains depend on it
+        if len(self.domains) > 0:
```

Comment:
One certificate is attached to many domains, especially true when the certificate has SAN. When domain was detach it was deleting the secret.

10 domains, 1 certificate. Detach one domain and then the secret is gone for all the other ones.

I thought about doing it the Pythonic way but felt > 0 was more explicit, I can change it tho.